libbpf-rs/src/skeleton.rs

```diff
@@ -340,6 +340,36 @@ pub trait SkelBuilder<'a> {
 }
 
 /// A trait for opened skeleton.
+///
+/// In addition to the methods defined in this trait, skeletons that implement this trait will also
+/// have bespoke implementations of a few additional methods to facilitate access to global
+/// variables of the BPF program. These methods will be named `bss()`, `data()`, and `rodata()`.
+/// Each corresponds to the variables stored in the BPF ELF program section of the same name.
+/// However if your BPF program lacks one of these sections the corresponding rust method will not
+/// be generated.
+///
+/// The type of the value returned by each of these methods will be specific to your BPF program.
+/// A common convention for BPF CO-RE programs is to define a single global variable in the BPF
+/// program with a struct type containing a field for each configuration parameter
+/// <sup>\[[source]\]</sup>.  libbpf-rs auto-generates this pattern for you without you having to
+/// define such a struct type in your BPF program. It does this by examining each of the global
+/// variables in your BPF program's `.bss`, `.data`, and `.rodata` sections and then creating rust
+/// struct types `yourprogram_bss_types`, `yourprogram_data_types`, and `yourprogram_rodata_types`.
+/// Since these struct types are specific to the layout of your BPF program, they are not
+/// documented in this crate. However you can see documentation for them by running `cargo doc` in
+/// your own project and looking at the `imp` module. You can also view their implementation by
+/// looking at the `*.skel.rs` files in your `target` directory. To find the latest one you can run
+/// a command such as:
+///
+/// `find target -name '*.skel.rs' | xargs -- ls -1tr | tail -n1`
```

Comment:
Actually, one more thing. The generated `*.skel.rs` being located in the target directory is a property of your build script. You can specify any destination directory. Similarly, if you use `cargo libbpf gen` for the generation, the generated skeleton will reside next to your BPF C code, I believe. Let's either generalize this or just refer to the skeleton in a more abstract way. I don't want to include information in an authoritative sounding tone that really depends on stuff the developer does.